packages/block-library/src/navigation/edit/index.js

```diff
@@ -263,7 +260,13 @@ function Navigation( {
 		 */
 		__unstableMarkNextChangeAsNotPersistent();
 		setRef( fallbackNavigationMenus[ 0 ].id );
-	}, [ navigationMenus ] );
+	}, [
+		navigationMenus,
```

Comment:
`navigationMenus` doesn't seem to be used in the effect. It's curious. Guessing it was here because other values are derived from it.

The `useEffect` after this also seems problematic.

```suggestion
```

Reply:
Great spot 👍